scripts/scraping/ddr-world.mjs

```diff
@@ -1,302 +1,379 @@
-  }
-}
+    while (emptyPageCount < maxEmptyPages) {
+      const offset = currentPage;
 
-/**
- * Name conversion mapping
- * Maps DDR World display names to standardized names
- * Can be used for both song names and artist names
- */
-const NAME_CONVERSIONS = {
-  ＴЁЯＲＡ: "TЁЯRA",
-  "Blind Justice ～Torn souls, Hurt Faiths ～":
-    "Blind Justice ～Torn souls, Hurt Faiths～",
-  "BURNING HEAT！（3 Option MIX）": "BURNING HEAT! (3 Option MIX)",
-  "A-One  feat.Napoleon": "A-One feat. Napoleon",
-  "GRADIUS REMIX（↑↑↓↓←→←→BA Ver.)": "GRADIUS REMIX (↑↑↓↓←→←→BA Ver.)",
-  "Tatsh＆NAOKI": "Tatsh&NAOKI",
-  'BEMANI Sound Team "[x]"': 'BEMANI Sound Team "[𝑥]"',
-};
+      // Construct URL with offset parameter
+      const url = new URL(this.#songListUrl);
+      url.searchParams.set("offset", offset.toString());
+      const pageUrl = url.toString();
 
-/**
- * Normalize name using conversion mapping
- * @param {string} name Original name (song name or artist name)
- * @returns {string} Normalized name
- */
-function normalizeName(name) {
-  let normalized = name;
-
-  // Apply all defined conversions
-  for (const [original, replacement] of Object.entries(NAME_CONVERSIONS)) {
-    // Escape special regex characters in the original string
-    const escapedOriginal = original.replace(/[.*+?^${}()|[\]\\]/g, "\\$&");
-    normalized = normalized.replace(
-      new RegExp(escapedOriginal, "g"),
-      replacement,
-    );
-  }
+      console.log(`Fetching page ${currentPage + 1}... (offset=${offset})`);
 
-  return normalized;
-}
+      try {
+        const pageSongs = await scrape(pageUrl);
 
-/**
- * @typedef {Pick<ChartData, "style" | "diffClass" | "lvl">} Chart
- */
+        if (pageSongs.length === 0) {
+          emptyPageCount++;
+          console.log(
+            `Page ${currentPage + 1} is empty (consecutive empty pages: ${emptyPageCount}/${maxEmptyPages})`,
+          );
+        } else {
+          emptyPageCount = 0; // Reset counter when songs are found
+          allSongs.push(...pageSongs);
+          console.log(
+            `Page ${currentPage + 1}: ${pageSongs.length} songs fetched (total: ${allSongs.length} songs)`,
+          );
 
-/**
- * @typedef {Pick<SongData, "name" | "artist" | "saHash" | "charts">} Song
- */
+          // If songs fetched is less than songsPerPage, likely the last page
+          if (pageSongs.length < songsPerPage) {
+            console.log(
+              `Songs fetched (${pageSongs.length}) is less than expected (${songsPerPage}), assuming last page`,
+            );
+            break;
+          }
+        }
 
-/**
- * Scrapes song data from DDR World official website
- * @param {string} pageUrl DDR World song list URL
- * @returns {Promise<Song[]>}
- */
-export async function getSongsFromDDRWorld(pageUrl) {
-  const dom = await getDom(pageUrl);
-  if (!dom) return [];
+        // Wait briefly between pages (reduce server load)
+        await new Promise((resolve) => setTimeout(resolve, 1000));
```

Comment:
just a note: the way I've avoided adding lots of server load is with the `requestQueue` in `scripts/utils.mjs`. You're already using the `getDom` method from that file, which is internally passing all requests for pages through that queue, so your page requests are actually already being throttled by the throughput limits set on the PQueue options around line 102